src/components/RouteOptionList/RouteOptionList.jsx

```diff
@@ -1,15 +1,51 @@
-import React from "react";
+// import React from "react";
+// import PropTypes from "prop-types";
+// import RouteOption from "../RouteOption/RouteOption";
+
+// export default function RouteOptionList({routeOptionProps}) {
+//   return (
+//     <div>
+//       {routeOptionProps.map(props => RouteOption(props))}
+//     </div>
+//   );
+// }
+
+// RouteOptionList.propTypes = {
+//   routeOptionProps: PropTypes.arrayOf(RouteOption.propTypes),
+// }
+
+
+
+
+import React, { useState } from "react";
 import PropTypes from "prop-types";
 import RouteOption from "../RouteOption/RouteOption";
 
-export default function RouteOptionList({routeOptionProps}) {
+export default function RouteOptionList({ routeOptionProps, onRouteOptionSelected }) {
+  const [selectedIndex, setSelectedIndex] = useState(null);
+
+  const handleOptionClick = (index, providerId, routeType) => {
+    setSelectedIndex(index);
+    if (onRouteOptionSelected) {
+      onRouteOptionSelected(providerId, routeType);
+    }
+  };
+
   return (
     <div>
-      {routeOptionProps.map(props => RouteOption(props))}
+      {routeOptionProps.map((props, index) => (
+        <RouteOption
+          key={index}
+          {...props}
+          isSelected={index === selectedIndex}
+          onClick={() => handleOptionClick(index, props.provider.name, props.type)}
+        />
+      ))}
     </div>
   );
 }
 
 RouteOptionList.propTypes = {
-  routeOptionProps: PropTypes.arrayOf(RouteOption.propTypes),
-}
+  routeOptionProps: PropTypes.arrayOf(PropTypes.shape(RouteOption.propTypes)),
+  onRouteOptionSelected: PropTypes.func.isRequired,
+};
```

Comment:
nitpick - missing EOL

Reply:
done in https://github.com/Micro-Commute/micro-commute-ui/pull/18/commits/4445278a65758836e3c7a7fbc34db7456b812ed9